federation-2/router-bridge/src/validate.rs

```diff
@@ -0,0 +1,72 @@
+/*!
+# Run introspection against a GraphQL schema and obtain the result
+*/
+
+use crate::error::Error;
+use crate::js::Js;
+use serde::{Deserialize, Serialize};
+use std::fmt::Display;
+use thiserror::Error;
+
+/// An error which occurred during JavaScript validation.
+///
+/// The shape of this error is meant to mimick that of the error created within
+/// JavaScript, which is a [`GraphQLError`] from the [`graphql-js`] library.
+///
+/// [`graphql-js']: https://npm.im/graphql
+/// [`GraphQLError`]: https://github.com/graphql/graphql-js/blob/3869211/src/error/GraphQLError.js#L18-L75
+#[derive(Debug, Error, Serialize, Deserialize, PartialEq, Eq, Clone)]
+pub struct ValidationError {
+    /// A human-readable description of the error that prevented introspection.
+    pub message: Option<String>,
+}
+
+impl Display for ValidationError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str(self.message.as_deref().unwrap_or("UNKNOWN"))
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
+pub struct ValidationResponse {
+    /// The introspection response if batch_introspect succeeded
+    #[serde(default)]
+    #[serde(rename = "Ok")]
+    data: Option<serde_json::Value>,
+    /// The errors raised on this specific query if any
+    #[serde(default)]
+    #[serde(rename = "Err")]
+    errors: Option<Vec<ValidationError>>,
```

Comment:
```suggestion
    pub data: Option<serde_json::Value>,
    /// The errors raised on this specific query if any
    #[serde(default)]
    #[serde(rename = "Err")]
    pub errors: Option<Vec<ValidationError>>,
```

I think these need to be pub so apollo-rs fuzzing can read them